src/components/DataDictionary/common/utils.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  Attribute,
+  Class,
+  DataDictionary,
+  DataDictionaryAnnotation,
+} from "../../../common/entities";
+import { SiteConfig } from "../../../config/entities";
+
+/**
+ * Annotate each entity column configuration with data dictionary values. Specifically,
+ * look up label and description for each column key.
+ * @param siteConfig - Site configuration to annotate.
+ * @param annotationsByKey - Data dictionary annotations keyed by key.
+ */
+export function annotateColumnConfig(
+  siteConfig: SiteConfig,
+  annotationsByKey: Record<string, DataDictionaryAnnotation>
+): void {
+  // Annotate every column in every entity.
+  siteConfig.entities.forEach((entity) => {
+    entity.list.columns.forEach((columnConfig) => {
+      // Find the annotation for the column key.
+      const annotation = annotationsByKey[columnConfig.id];
+      if (!annotation) {
+        return;
+      }
+
+      if (!columnConfig.meta) {
+        columnConfig.meta = {};
+      }
+      columnConfig.meta.annotation = annotation;
+    });
+  });
+}
+
+/**
+ * Annotate filter and colummn configuration with data dictionary values. Note this
+ * functionality mutates the site config. A possible future improvement would be to
+ * create either a specific "raw"  or "annotated" type to indicate clearly the point
+ * at which the config has been annotated.
+ * @param siteConfig - The site configuration to annotate.
+ */
+export function annotateSiteConfig(siteConfig: SiteConfig): void {
+  // Build and map data dictionary annotations by key.
+  const { dataDictionary } = siteConfig;
+  if (!dataDictionary) {
+    return;
+  }
+  const annotationsByKey = keyAnnotationsByKey(dataDictionary);
+
+  // Annotate elements of site config.
+  annotateEntityConfig(siteConfig, annotationsByKey);
+  annotateCategoryConfig(siteConfig, annotationsByKey);
```

Comment:
Do we also need to annotate `entities.categoryGroupConfig`?

Reply:
I might be missing something here! Why are you thinking `entities.categoryGroupConfig` also needs annotating?

Reply:
If `entities.categoryGroupConfig` is defined, it overrides `config.categoryGroupConfig`. This is useful when entity categories differ, such as in Tracker Atlases and Tracker Reports.

Reply:
Got it, thanks!